pkg/frontend/transport/handler.go

```diff
@@ -117,6 +124,9 @@ func NewHandler(cfg HandlerConfig, roundTripper http.RoundTripper, log log.Logge
 }
 
 func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	activityIndex := f.at.Insert(httpRequestActivityGenerator(r))
```

Comment:
Can we follow the same pattern as we use elsewhere when inserting activities:

```suggestion
	activityIndex := f.at.Insert(func() string {
		return httpRequestActivity(r)
	})
```

Reply:
I can't see the advantage of making this function longer with the same functionality.

Reply:
Consistency with other call sites. And reducing indirection in `httpRequestActivityGenerator` -- no need to return function there.

Reply:
But then we're creating an ad-hoc function here. I'd rather replace other places to do it like this :thinking:

Reply:
It isn't such a strong preference here for me, so changing in https://github.com/grafana/mimir/pull/3561/commits/601cc2fed552848e937a6132d83c3a1d7c8696ab

Reply:
> But then we're creating an ad-hoc function here. I'd rather replace other places to do it like this 🤔

Yes, we need that function somewhere. Replacing other places is an option too I guess, but for a separate PR :) I personally prefer this version, as I can more easily see how it works, but it doesn't matter too much.